File: back/DevC/src/main/java/com/back/devc/global/initData/AdminAccountInitData.java
package com.back.devc.global.initData;

import com.back.devc.domain.member.member.entity.Member;
import com.back.devc.domain.member.member.repository.MemberRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.ApplicationRunner;
import org.springframework.context.annotation.Bean;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class AdminAccountInitData {

    private static final String ADMIN_EMAIL = "admin@test.com";
    private static final String ADMIN_PASSWORD = "admin123@";
    private static final String ADMIN_NICKNAME_BASE = "admin";

    private final MemberRepository memberRepository;
    private final PasswordEncoder passwordEncoder;

    @Bean
    public ApplicationRunner adminAccountInitRunner() {
        return args -> {
            if (memberRepository.existsByEmail(ADMIN_EMAIL)) {
                return;
            }

            String encodedPassword = passwordEncoder.encode(ADMIN_PASSWORD);
            String nickname = resolveUniqueNickname(ADMIN_NICKNAME_BASE);

            Member admin = Member.createLocalAdminMember(
                    ADMIN_EMAIL,
                    encodedPassword,
                    nickname
            );

            memberRepository.save(admin);
        };
    }

    private String resolveUniqueNickname(String base) {
        if (!memberRepository.existsByNickname(base)) {
            return base;
        }

        int sequence = 1;
        String candidate = base + sequence;
        while (memberRepository.existsByNickname(candidate)) {
            sequence++;
            candidate = base + sequence;
        }
        return candidate;
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/global/initData/AdminAccountInitData.kt
package com.back.devc.global.initData

import com.back.devc.domain.member.member.entity.Member
import com.back.devc.domain.member.member.repository.MemberRepository
import org.springframework.boot.ApplicationRunner
import org.springframework.context.annotation.Bean
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Component

@Component
class AdminAccountInitData(
    private val memberRepository: MemberRepository,
    private val passwordEncoder: PasswordEncoder
) {

    @Bean
    fun adminAccountInitRunner(): ApplicationRunner {
        return ApplicationRunner {
            if (memberRepository.existsByEmail(ADMIN_EMAIL)) {
                return@ApplicationRunner
            }

            val encodedPassword = passwordEncoder.encode(ADMIN_PASSWORD)
                ?: throw IllegalStateException("관리자 비밀번호 인코딩에 실패했습니다.")

            val nickname = resolveUniqueNickname(ADMIN_NICKNAME_BASE)

            val admin = Member.createLocalAdminMember(
                ADMIN_EMAIL,
                encodedPassword,
                nickname
            )

            memberRepository.save(admin)
        }
    }

    private fun resolveUniqueNickname(base: String): String {
        if (!memberRepository.existsByNickname(base)) {
            return base
        }

        var sequence = 1
        var candidate = base + sequence
        while (memberRepository.existsByNickname(candidate)) {
            sequence++
            candidate = base + sequence
        }

        return candidate
    }

    companion object {
        private const val ADMIN_EMAIL = "admin@test.com"
        private const val ADMIN_PASSWORD = "admin123@"
        private const val ADMIN_NICKNAME_BASE = "admin"
    }
}